.github/workflows/master.yml

```diff
@@ -11,6 +11,10 @@ jobs:
   check:
     name: Check
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        feature: [libp2p, waku]
```

Comment:
It seems the `fail-fast: true` doesn't work because we have the `continue-on-error: true` for the `cargo check` step below?
For example, the `cargo check` failed in the PR: https://github.com/logos-co/nomos-node/actions/runs/6046982217/job/16409567703?pr=316, but it show me the green light.


<img width="1728" alt="image" src="https://github.com/logos-co/nomos-node/assets/5462944/63b33b20-03bb-4604-bcde-b1f02f0270ef">

Reply:
`fail-fast: true` is set for master branch. PRs has a separate file `prs.yml` and it's set to `fail-fast: false`. But I'll update the cargo check step to `continue-on-error: false`

Reply:
Ah, i left this comment at the wrong position :)) but it seems the `pr.yml` also has the same settings.